examples/leds_shell/main.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2024 Krzysztof Cabaj <kcabaj@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       leds_shell - sample application for demonstrating internal
+ *              board LEDs on/off and basic GPIO using interactive RIOT shell
+ *
+ * @author      Krzysztof Cabaj <kcabaj@gmail.com>
+ *
+ * @}
+ */
+
+#include "stdio.h"
+#include "stdlib.h"
+#include "shell.h"
+#include "led.h"
+#include <periph/gpio.h>
+
+static int gpio_command(int argc, char **argv)
+{
+    if (argc < 4) {
+        printf("usage: %s <init/set/clear> <port no.> <pin no.>\n", argv[0]);
+        return -1;
+    }
+
+    int port_no = atoi(argv[2]);
+    int pin_no = atoi(argv[3]);
+
+    if (strcmp(argv[1], "init") == 0) {
+        printf("GPIO initialization PORT %d, PIN %d\n", port_no, pin_no);
+
+        int result;
+
+        result = gpio_init(GPIO_PIN(port_no, pin_no), GPIO_OUT);
+
+        if (result == 0) {
+            printf("Success!\n");
+        }
+        else if (result == -1) {
+            printf("Failure!\n");
+        }
```

Comment:
While it is correct as per our API, I think it is better to fallback to a plain else here (no else if)
For instance, rp2040 gpio driver might return `-ENOTSUP`. Of course, this is the gpio driver that doesn't match the documentation but better safe than sorry here.